HighwayPatrol.cpp

```diff
@@ -27,8 +32,12 @@ void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
     std::cout << name << ": vehicle is traveling " << v->speed - h->speedLimit << " miles per hour over the speed limit" << std::endl;
     if( willArrest )
     {
-        //print the vehicle type in this std::cout between "THE [" and "] PULL". 
-        std::cout << name << ": YOU IN THE [ " << " ] PULL OVER AND SHOW YOUR HANDS" << std::endl;
+        //print the vehicle type in this std::cout between "THE [" and "] PULL".
+        if( auto* t = dynamic_cast<Car*>(v) ) type = "CAR";
+        if( auto* t = dynamic_cast<Motorcycle*>(v) ) type = "MOTORCYCLE";
+        // if( auto* t = dynamic_cast<SemiTruck*>(v) ) std::string s = "TRUCK";
```

Comment:
if it's a car, don't check if it's a truck or motorcycle.  you're still checking if it is, for all of the 3 choices.
use `else if`
Also, trucks can still speed, so don't omit them from this check.
- [ ] revise